test_scripts/Polices/build_options/046_ATF_P_TC_PTU_UTF8_Encoding_Check_HTTP.lua

```diff
@@ -114,12 +114,19 @@ require("user_modules/AppTypes")
 
 --[[ Specific Notifications ]]
 function Test:RegisterNotification()
+  local request_http_received = false
   self.mobileSession:ExpectNotification("OnSystemRequest")
   :Do(function(_, d)
-      ptu = json.decode(d.binaryData)
+      if(d.payload.requestType == "HTTP") then
+        request_http_received = true
+        ptu = json.decode(d.binaryData)
+      end
     end)
-  :Times(AtLeast(1))
-  :Pin()
+  :Times(2)
+
+  if(request_http_received == false) then
```

Comment:
@istoimenova, Due to asynchronous registering of expectations (in ATF) test case will be failed before 2nd OnSystemRequest received
One of the solutions is to implement verification in other test step